torch/_functorch/functional_call.py

```diff
@@ -77,3 +77,52 @@ def functional_call(
         parameters_and_buffers = {k: v for d in parameter_and_buffer_dicts for k, v in d.items()}
 
     return nn.utils.stateless.functional_call(module, parameters_and_buffers, args, kwargs)
+
+
+@exposed_in("torch.func")
+def stack_ensembled_state(models):
+    """stack_ensembled_state(models) -> params, buffers
+
+    Prepares a list of torch.nn.Modules for ensembling with :func:`vmap`.
+
+    Given a list of ``M`` ``nn.Modules`` of the same class, returns two dictionaries
+    that stack all of their parameters and buffers together, indexed by name.
+
+    Here's an example of how to ensemble over a very simple model:
+
+    .. code-block:: python
+
+        num_models = 5
+        batch_size = 64
+        in_features, out_features = 3, 3
+        models = [torch.nn.Linear(in_features, out_features) for i in range(num_models)]
+        data = torch.randn(batch_size, 3)
+
+        def wrapper(params, buffers, data):
+            return functorch.functional_call(model[0], (params, buffers), data)
+
+        params, buffers = stack_ensembled_state(models)
+        output = vmap(wrapper, (0, 0, None))(params, buffers, data)
+
+        assert output.shape == (num_models, batch_size, out_features)
+
+    .. warning::
+        All of the modules being stacked together must be the same (except for
+        the values of their parameters/buffers). For example, they should be in the
+        same mode (training vs eval).
+    """
+    if len(models) == 0:
+        raise RuntimeError('stack_ensembled_state: Expected at least one model, got 0.')
+    if not (all(m.training for m in models) or all(not m.training for m in models)):
+        raise RuntimeError('stack_ensembled_state: Expected all models to '
+                           'have the same training/eval mode.')
+    model0_typ = type(models[0])
+    if not all(type(m) == model0_typ for m in models):
+        raise RuntimeError('stack_ensembled_state: Expected all models to '
+                           'be of the same class.')
```

Comment:
I might not be looking in the right places, but, are there tests that check these errors get hit?

Reply:
Somehow these test got added to a different commit so thanks for calling it out. They're now in `test_stack_module_state_error` in `test_eager_transforms.py`